wrangler-service/src/main/java/io/cdap/wrangler/service/bigquery/BigQueryHandler.java

```diff
@@ -429,6 +397,41 @@ public void specification(HttpServiceRequest request, HttpServiceResponder respo
     return new Pair<>(rows, schemaToReturn);
   }
 
+  private Object getRowValue(StandardSQLTypeName standardType, FieldValue fieldValue) {
+    switch (standardType) {
+      case TIME:
+        return LocalTime.parse(fieldValue.getStringValue());
+      case DATE:
+        return LocalDate.parse(fieldValue.getStringValue());
+      case TIMESTAMP:
+        long tsMicroValue = fieldValue.getTimestampValue();
+        return getZonedDateTime(tsMicroValue);
+      case NUMERIC:
+        BigDecimal decimal = fieldValue.getNumericValue();
+        if (decimal.scale() < 9) {
+          // scale up the big decimal. this is because structured record expects scale to be exactly same as schema
+          // Big Query supports maximum unscaled value up to 38 digits. so scaling up should still be <= max
+          // precision
+          decimal = decimal.setScale(9);
+        }
+        return decimal;
+
+      case DATETIME:
```

Comment:
unrelated note: it seems like we need to support a datetime type in CDAP, otherwise there is no way the BQ sink can create a table with datetime column type. For example, if somebody wants to copy the contents of one BQ table to another BQ table. Can you open a jira for this?

Reply:
Issue: https://issues.cask.co/browse/CDAP-15370